conn-sdk-cli/readmegen/preprocess.go

```diff
@@ -0,0 +1,126 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package readmegen
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+const (
+	readmegenTagPrefix = "readmegen:"
+	readmegenOpenTag   = "<!-- " + readmegenTagPrefix
+	readmegenCloseTag  = "<!-- /" + readmegenTagPrefix
+
+	readmegenOpenTagLength  = len(readmegenOpenTag)
+	readmegenCloseTagLength = len(readmegenCloseTag)
+
+	commentClose       = "-->"
+	commentCloseLength = len(commentClose)
+)
+
+var (
+	preprocessTags = map[string]string{
+		"name":        `{{ title .specification.name }}`,
+		"summary":     `{{ .specification.summary }}`,
+		"description": `{{ .specification.description }}`,
+		"version":     `{{ .specification.version }}`,
+		"author":      `{{ .specification.author }}`,
+
+		"source.parameters.yaml":       `{{ template "parameters.yaml" args "specification" .specification "parameters" .specification.source.parameters }}`,
+		"source.parameters.table":      `{{ template "parameters.table" args "specification" .specification "parameters" .specification.source.parameters }}`,
+		"destination.parameters.yaml":  `{{ template "parameters.yaml" args "specification" .specification "parameters" .specification.destination.parameters }}`,
+		"destination.parameters.table": `{{ template "parameters.table" args "specification" .specification "parameters" .specification.destination.parameters }}`,
+	}
+	errReadmegenCommentNotFound = errors.New("readmegen open tag not found")
+)
+
+// Preprocess takes the contents of a readme file and preprocesses it by replacing
+// readmegen tags with the corresponding values. A readmegen tag is a HTML comment
```

Comment:
"replacing readmegen tags with the corresponding values" makes me think that Preprocess produces a string with values, but it actually transforms the readme with tags into a readme Go template?

Reply:
Rephrased `values` to `Go templates` 👍